tests/Unit/NumericalAlgorithms/LinearOperators/Test_PartialDerivatives.cpp

```diff
@@ -648,6 +651,341 @@ void test_partial_derivatives_spherical_shell() {
+
+  const size_t dv_size = get<0>(coords).size();
+
+  auto& scalar = get<::Tags::TempScalar<0>>(prefactor_vars);
+  auto& d_scalar = get<::Tags::Tempi<0, 3>>(d_prefactor_vars);
+  // scalar, doing nothing
+  scalar.get() = DataVector(dv_size, 1.0);
+  // \partial_i of 1 is zero
+  for (size_t i = 0; i < index_dim<0>(d_scalar); ++i) {
+    d_scalar.get(i) = DataVector(dv_size, 0.0);
+  }
+
+  auto& one_form = get<::Tags::Tempa<0, 3>>(prefactor_vars);
+  auto& d_one_form = get<::Tags::Tempia<0, 3>>(d_prefactor_vars);
+  auto& vector = get<::Tags::TempA<0, 3>>(prefactor_vars);
+  auto& d_vector = get<TempiA>(d_prefactor_vars);
+  if constexpr (spherical) {
+    // spherical case, vector/one form, using x^a, initializing time component
+    // to zero
+    get<0>(vector) = get<0>(one_form) = DataVector(dv_size, 0.0);
+    for (size_t i = 1; i < index_dim<0>(vector); ++i) {
+      vector.get(i) = one_form.get(i) = coords.get(i - 1);
+    }
+    // partial_i of x_a is \delta_ia
+    for (size_t i = 0; i < index_dim<0>(d_vector); ++i) {
+      for (size_t a = 0; a < index_dim<1>(d_vector); ++a) {
+        if ((i + 1) == a and a != 0) {
+          d_vector.get(i, a) = d_one_form.get(i, a) = DataVector(dv_size, 1.0);
+        } else {
+          d_vector.get(i, a) = d_one_form.get(i, a) = DataVector(dv_size, 0.0);
+        }
+      }
+    }
+  } else {
+    // axial case, vector/one form, using x^a = (0, -z, 0, x) (pure rotation)
+    get<0>(vector) = get<0>(one_form) = DataVector(dv_size, 0.0);
+    get<1>(vector) = get<1>(one_form) = -1.0 * coords.get(2);
+    get<2>(vector) = get<2>(one_form) = DataVector(dv_size, 0.0);
+    get<3>(vector) = get<3>(one_form) = get<0>(coords);
+
+    // \partial_i of x_a is (0, \delta_i2, 0, \delta_i0)
+    for (size_t i = 0; i < index_dim<0>(d_vector); ++i) {
+      for (size_t a = 0; a < index_dim<1>(d_vector); ++a) {
+        if (i == 2 and a == 1) {
+          d_vector.get(i, a) = d_one_form.get(i, a) = DataVector(dv_size, -1.0);
+        } else if (i == 0 and a == 3) {
+          d_vector.get(i, a) = d_one_form.get(i, a) = DataVector(dv_size, 1.0);
+        } else {
+          d_vector.get(i, a) = d_one_form.get(i, a) = DataVector(dv_size, 0.0);
+        }
+      }
+    }
+  }
+
+  auto& mixed = get<::Tags::TempAb<0, 3>>(prefactor_vars);
+  auto& d_mixed = get<TempiAb>(d_prefactor_vars);
+  auto& rank2 = get<::Tags::Tempab<0, 3>>(prefactor_vars);
+  auto& d_rank2 = get<Tempiab>(d_prefactor_vars);
+  // filling with 0's in time components
+  // filling space with (essenially) projector to tangent space of sphere
+  // P_ab = \delta_ab + x_a x_b
+  // can have arbitrary function in from of each term, not doing here
+  // (the real projector is \delta_ij - x_i x_j / r^2)
+  for (size_t a = 0; a < index_dim<0>(rank2); ++a) {
+    for (size_t b = 0; b < index_dim<1>(rank2); ++b) {
+      if (a == 0 or b == 0) {
+        mixed.get(a, b) = rank2.get(a, b) = DataVector(dv_size, 0.0);
```

Comment:
Does this mean you aren't testing e.g. `d_i g_{tt}`? Because I feel like that's one we should make sure works